python/lsst/ip/isr/photodiode.py

```diff
@@ -265,3 +276,31 @@ def integrateTrimmedSum(self):
         lowValueIndices = np.where(self.currentSamples < currentThreshold)
         baseline = np.median(self.currentSamples[lowValueIndices])
         return np.trapz(self.currentSamples - baseline, self.timeSamples)
+
+    def integrateChargeSum(self):
+        """For this method, the values in .currentSamples are actually the
+        integrated charge values as measured by the ammeter for each
+        sampling interval.  We need to do a baseline subtraction,
+        based on the charge values when the LED is off, then sum up
+        the corrected signals.
+
+        Returns
+        -------
+        sum : `float`
+            Total charge measured.
+        """
+        dt = self.timeSamples[:, 1:] - self.timeSamples[:, :-1]
+        # The .currentSamples values are the current integrals over
+        # the interval preceding the current time stamp, so omit the
+        # first value.
+        charge = self.currentScale*self.currentSamples[:, 1:]
```

Comment:
I don't think I understand this scaling.  From the two example photodiode files I've looked at, the current samples are all positive, so this inverts them all to be negative.  I think that and the min/maxes below inverts the measurements, such that with my test photodiode (instrument='LSSTCam', exposure=3021120700810, collections="LSSTCam/photodiode" as it was the first to return a result from the butler) yields `bg_current=-1.19e-7`, `return=3.50e-7`, and `np.sum(charge)=-1.12e-6` `np.sum(-bg_current*dt)=1.47e-6`.  The `index` for my example also is selecting the points where `current ~ -1.2e-7`, while `~index` selects those around -1e-12.  
Have I somehow confused myself?

Reply:
The data from the CCOB uses a different ammeter that for the BOT data, and the sampled points have negative values for the intervals where the CCOB LED is on, and so differs from the BOT data in that regard.  See instrument='LSST-TS8', exposure=202305122056562, collections="LSST-TS8/photodiode" in /repo/ir2.

Reply:
Ok, I was confused about the inputs then, and this makes the default value make more sense to me.  The test TS8 photodiode I tried to look at must not have been ingested for some reason.
With a proper example, the values come out looking like I was expecting the first time, so I think that clears my confusion.